pkg/middleware/log_server_error_request_test.go

```diff
@@ -0,0 +1,100 @@
+package middleware
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	ce "github.com/content-services/content-sources-backend/pkg/errors"
+	"github.com/labstack/echo/v4"
+	"github.com/stretchr/testify/assert"
+)
+
+const TestBody = `{"limit": 100, "search": "random", "uuids": ["e88efd75-2b29-4b59-8867-bb60435b3742"]}`
+const LargeBody = `{"limit": 100, "search": "Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Integer pellentesque quam vel velit. Vivamus porttitor turpis ac leo. Ut tempus purus at lorem. Nullam justo enim, consectetuer nec, ullamcorper ac, vestibulum in, elit. Nulla pulvinar eleifend sem. Proin mattis lacinia justo. Integer malesuada. Etiam sapien elit, consequat eget, tristique non, venenatis quis, ante. Etiam dui sem, fermentum vitae, sagittis id, malesuada in, quam. Integer lacinia. Proin pede metus, vulputate nec, fermentum fringilla, vehicula vitae, justo. Curabitur bibendum justo non orci. Ut enim ad minima veniam, quis nostrum exercitationem ullam corporis suscipit laboriosam, nisi ut aliquid ex ea commodi consequatur? Etiam sapien elit, consequat eget, tristique non, venenatis quis, ante. Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas. Maecenas sollicitudin. Quisque porta. Nullam sapien sem, ornare ac, nonummy non, lobortis a enim. Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas. Maecenas sollicitudin. Quisque porta. Nullam sapien sem, ornare ac, nonummy non, lobortis a enim.", "uuids": ["e88efd75-2b29-4b59-8867-bb60435b3742"]}`
+
+func TestReadBodyStoreAndRestore(t *testing.T) {
+	e := echo.New()
+	req := httptest.NewRequest(http.MethodGet, "/", bytes.NewBufferString(TestBody))
```

Comment:
small thing, since GET requests don't usually have a body, maybe we should check for both GET and POST requests (or just POST) in the tests? it could be good to test what's logged for requests with no body too. right now for requests with no body i see this: 
`11:34AM ERR ../../go/pkg/mod/github.com/content-services/lecho/v3@v3.5.2/logger.go:88 > Request body:  request_id=a80aff68-f533-4677-a628-b245295ac4b0`

Reply:
oh, must have missed that, for clarity in the tests it probably should be a `POST` or a method usually having a body (not that it would change the outcome/functionality in the case of these tests 😅)
that gets me thinking, if I should add a check for the request method and if it's `GET` then don't log this message at all 🤔

Reply:
[resolved in the latest commit](https://github.com/content-services/content-sources-backend/pull/773/commits/8a34ffb865d15367b1bdda71ef5420a7bcab6422), turned off logging for `GET` and `DELETE` methods as they normally shouldn't have a body